src/IIIFAuth2/IIIFAuth2.API/Utils/GuardX.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace IIIFAuth2.API.Utils;
+
+public static class GuardX
+{
+    /// <summary>
+    /// Throw <see cref="ArgumentNullException"/> if provided object is null.
+    /// </summary>
+    /// <param name="argument">Argument to check.</param>
+    /// <param name="argName">Name of argument.</param>
+    /// <typeparam name="T">Type of argument to check.</typeparam>
+    /// <returns>Passed argument, if not null.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if provided argument is null.</exception>
+    [return: NotNull]
+    public static T ThrowIfNull<T>(this T argument, string argName)

```

Comment:
Usage of this extension method is slightly different than `ArgumentNullException.ThrowIfNull()` as the above returns the object if it's not null. Allowing, for example, usage when assigning

```cs
var thing = myObject.Property.ThrowIfNull("xx");
thing.Whatever();
```

Rather than
```cs
ArgumentNullException.ThrowIfNull(myObject.Property, "xx");
var thing = myObject.Property;
thing.Whatever();
```